### PR TITLE
[CMake] Add icpx/icx compiler support.

### PR DESCRIPTION
**Pull request by bader**

### CODE REVIEW

**Review by wanyaman — APPROVED** [review]
LGTM

**Comment by wanyaman** [comment]
@bader @Can you merge PR after getting approval from code owners?

**Comment by bader** [comment]
> @bader @can you merge PR after getting approval from code owners?

Yes, I'll merge.

@ProGTX, @keryell, @illuhad, do you have any comments for this change?

**Review by psalz — APPROVED** [review]

**Pull request merged**
